README.md

```diff
@@ -116,8 +116,9 @@ independent top-level definitions.
 
 * CPP support is experimental. CPP is virtually impossible to handle
   correctly, so we process them as a sort of unchangeable snippets. This
-  works only in simple cases when CPP conditionals are self-contained. Use
-  Ormolu with CPP at your own risk.
+  works only in simple cases when CPP conditionals surround top-level
+  declarations. See the [CPP][design-cpp] section in the design notes for
+  a discussion of the dangers.
```

Comment:
I made an attempt to be more precise here. Maybe ormolu could have some check of CPP directives really surrounding top-level declarations. I'm rather thinking aloud.

Reply:
Many times it also works in the situations when CPP is in depth of a definitions, for example her it worked correctly: https://github.com/mrkkrp/zip/blob/master/Codec/Archive/Zip/Internal.hs#L566-L596. Of course it's just luck.